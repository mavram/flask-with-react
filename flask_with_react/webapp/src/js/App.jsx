
import React from 'react'
import {BrowserRouter, Route, Switch} from 'react-router-dom'
import {SettingsComponent} from './containers/SettingsComponent'
import {Index, Login, Logout} from './views/Index'
import {Home} from './views/Home'
import {Welcome} from './views/Welcome'


export const App = () => {
    return <BrowserRouter>
        <SettingsComponent>
            <Switch>
                <Route exact path='/' render={Index}/>
                <Route exact path='/home' render={Home}/>
                <Route exact path='/welcome' render={Welcome}/>
                <Route exact path='/login' render={Login}/>
                <Route exact path='/logout' render={Logout}/>
            </Switch>
        </SettingsComponent>
    </BrowserRouter>
}
