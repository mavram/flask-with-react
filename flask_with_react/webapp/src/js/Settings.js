
export const JWT_KEY = '__jwt_key__'

export class Settings {
    static getJWT() {
        return localStorage.getItem(JWT_KEY)
    }

    static isAuthenticated() {
        return this.getJWT()
    }

    static set(name, value) {
        value ? localStorage.setItem(name, value) : localStorage.removeItem(name)
    }
}
