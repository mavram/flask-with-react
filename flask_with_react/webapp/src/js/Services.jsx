import {Service} from './modules/Service'
import {Settings} from './Settings'


export class LoginService extends Service {
    constructor() {
        super({
            name: 'login',
            verb: 'POST'
        })
    }

    login(username, passwd) {
        return this.call({jwt: Settings.getJWT(), body: {username: username, password: passwd}})
    }
}


export class ApplicationService extends Service {
    constructor() {
        super({
            name: 'application'
        })
    }

    get() {
        return this.call({jwt: Settings.getJWT()})
    }
}
