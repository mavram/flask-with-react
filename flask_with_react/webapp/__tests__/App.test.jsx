import React from 'react'
import renderer from 'react-test-renderer'
import {App} from '../src/js/App'


it('App basic creation', () => {
    const component = renderer.create(
        <App />
    )

    expect(component.toJSON()).toMatchSnapshot()
})
