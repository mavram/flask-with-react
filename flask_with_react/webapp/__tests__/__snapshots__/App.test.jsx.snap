// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App basic creation 1`] = `null`;
